plugins/modules/vmware_host_capability_info.py

```diff
@@ -192,10 +191,16 @@ def gather_host_capability_info(self):
                 supportedVmfsMajorVersion=[version for version in hc.supportedVmfsMajorVersion],
                 vmDirectPathGen2UnsupportedReason=[reason for reason in hc.vmDirectPathGen2UnsupportedReason],
                 ftCompatibilityIssues=[issue for issue in hc.ftCompatibilityIssues],
-                checkpointFtCompatibilityIssues=[issue for issue in hc.checkpointFtCompatibilityIssues],
                 smpFtCompatibilityIssues=[issue for issue in hc.smpFtCompatibilityIssues],
                 replayCompatibilityIssues=[issue for issue in hc.replayCompatibilityIssues],
             )
+
```

Comment:
Could you read `smpFtSupported` and `smpFtCompatibilityIssues` instead, see https://github.com/vmware/pyvmomi/issues/893#issuecomment-614878035

Reply:
Thank you for tell me that information :)
I understood that the parameters can be substituted as follows.

* `checkpointFtSupported` -> `smpFtSupported`
* `checkpointFtCompatibilityIssues` -> `smpFtCompatibilityIssues`

The `smpFtSupported` and `smpFtCompatibilityIssues` parameters exist as follows

https://github.com/ansible-collections/vmware/blob/3b03607c92e2cc7f2714287e8e961c70fa286643/plugins/modules/vmware_host_capability_info.py#L135
https://github.com/ansible-collections/vmware/blob/3b03607c92e2cc7f2714287e8e961c70fa286643/plugins/modules/vmware_host_capability_info.py#L194

So, I think I will remove the below lines.

https://github.com/ansible-collections/vmware/blob/3b03607c92e2cc7f2714287e8e961c70fa286643/plugins/modules/vmware_host_capability_info.py#L198-L202

Reply:
Hi @sky-joker, yes, we should also keep the compatibility with the previous model. I would do something like that:

```
if hasattr(hc, 'smpFtSupported'):
    hosts_capability_info[host.name]['checkpointFtSupported'] = hc.smpFtSupported
   hosts_capability_info[host.name]['smpFtSupported'] = hc.smpFtSupported
elif hasattr(hc, 'checkpointFtSupported'):
    hosts_capability_info[host.name]['checkpointFtSupported'] = hc.checkpointFtSupported
   hosts_capability_info[host.name]['smpFtSupported'] = hc.checkpointFtSupported
```

Reply:
okay :) 
I'll fix it that way.

Reply:
@goneri 

I'll amend it as follows, what do you think?

```python
if hasattr(hc, 'checkpointFtSupported'):
    # pyvmomi 6.7.x or less
    hosts_capability_info[host.name]['checkpointFtSupported'] = hc.checkpointFtSupported
else:
    # pyvmomi 7.x or more
    hosts_capability_info[host.name]['checkpointFtSupported'] = hosts_capability_info[host.name]['smpFtSupported']

if hasattr(hc, 'checkpointFtCompatibilityIssues'):
    # pyvmomi 6.7.x or less
    hosts_capability_info[host.name]['checkpointFtCompatibilityIssues'] = [issue for issue in hc.checkpointFtCompatibilityIssues]
else:
    # pyvmomi 7.x or more
    hosts_capability_info[host.name]['checkpointFtCompatibilityIssues'] = hosts_capability_info[host.name]['smpFtCompatibilityIssues']
```

Reply:
hum, this makes me wonder if we can we just simplify to?
```
hosts_capability_info[host.name]['checkpointFtSupported'] = hosts_capability_info[host.name]['smpFtSupported']
hosts_capability_info[host.name]['checkpointFtCompatibilityIssues'] = hosts_capability_info[host.name]['smpFtCompatibilityIssues']
```

It also would be nice to add a comment to explain the weird construction for these keys.

Reply:
```python
hosts_capability_info[host.name]['checkpointFtSupported'] = hosts_capability_info[host.name]['smpFtSupported']
hosts_capability_info[host.name]['checkpointFtCompatibilityIssues'] = hosts_capability_info[host.name]['smpFtCompatibilityIssues']
```

I think that's good :)